exchanges/bitfinex/bitfinex_wrapper.go

```diff
@@ -647,27 +647,36 @@ func (b *Bitfinex) SubmitOrder(o *order.Submit) (order.SubmitResponse, error) {
 
 // ModifyOrder will allow of changing orderbook placement and limit to
 // market conversion
-func (b *Bitfinex) ModifyOrder(action *order.Modify) (string, error) {
+func (b *Bitfinex) ModifyOrder(action *order.Modify) (order.Modify, error) {
 	if err := action.Validate(); err != nil {
-		return "", err
+		return order.Modify{}, err
 	}
 
 	orderIDInt, err := strconv.ParseInt(action.ID, 10, 64)
 	if err != nil {
-		return action.ID, err
+		return order.Modify{ID: action.ID}, err
 	}
 	if b.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
-		if action.Side == order.Sell && action.Amount > 0 {
-			action.Amount = -1 * action.Amount
-		}
-		err = b.WsModifyOrder(&WsUpdateOrderRequest{
+		request := WsUpdateOrderRequest{
 			OrderID: orderIDInt,
 			Price:   action.Price,
 			Amount:  action.Amount,
-		})
-		return action.ID, err
-	}
-	return "", common.ErrNotYetImplemented
+		}
+		if action.Side == order.Sell && action.Amount > 0 {
+			request.Amount *= -1
+		}
+		err = b.WsModifyOrder(&request)
+		return order.Modify{
```

Comment:
Can we dereference action pointer instead, then assign as needed?

Reply:
I'm not sure I understand you correctly. You want me to assign the change on the input Modify struct?

Reply:
What I mean was when you return we could just pass on a copy, so we could reduce some lines (up to you if you want to do it, resolve if not) e.g:
```go
return *action, nil 
// or 
re := *action
re.Price = 123
re.ID = 1234
return re, nil
```